src/scratchOrgSettingsGenerator.ts

```diff
@@ -53,6 +54,21 @@ export interface BusinessProcessFileContent extends JsonMap {
   ];
 }
 
+/** This is the contents of the package.xml that we will use when we deploy settings to a scratch org. */
+
+const packageFileContents = `<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
the package.xml structure is the same between source and mdapi format.  

Overall feedback: I strongly recommend using the more recent approach "build a js object that represents the file, then convert it to xml"

the file names will be different (source vs mdapi)
and what were separate files inside the object folder (record types) would now be nodes inside the object.

So, the approach would be:
create an interface that describes the parts of customObject that we care about: https://developer.salesforce.com/docs/atlas.en-us.236.0.api_meta.meta/api_meta/customobject.htmbps
write logic to populate those properties where needed for each CustomObject file
json->xml
write files